Classes/Controller/Application/AbstractApplicationController.php

```diff
@@ -75,6 +74,11 @@ public function initializeAction()
 
             $this->redirectToUri($loginUri);
         }
+
+        if (!$this->settings['loginPage'] && $this->request->hasArgument("application")) {
+            $GLOBALS['TSFE']->fe_user->setKey('ses', 'Application', (int)$this->request->getArgument("application"));
+            $GLOBALS["TSFE"]->storeSessionData();
```

Comment:
You can combine this with `setAndSaveSessionData()` instead. Also I'd recommend adding some kind of namespace to the `$key` since `Application` is rather generic and can lead to conflicts with other packages.